subgraphs/stablejoe/src/mappings/stablejoe.ts

```diff
@@ -0,0 +1,186 @@
+import { BigDecimal, BigInt, log } from '@graphprotocol/graph-ts';
+import {
+  BIG_DECIMAL_1E18,
+  BIG_DECIMAL_1E6,
+  BIG_DECIMAL_ZERO,
+  JOE_USDT_PAIR_ADDRESS,
+  USDC_ADDRESS,
+  USDT_ADDRESS
+} from 'const'
+import {
+  ClaimReward as ClaimRewardEvent,
+  Deposit as DepositEvent,
+  DepositFeeChanged as DepositFeeChangedEvent,
+  Withdraw as WithdrawEvent
+} from '../../generated/StableJoeStaking/StableJoeStaking'
+import {
+ StableJoe, 
+ User, 
+ StableJoeDayData
+} from '../../generated/schema'
+import { Pair as PairContract } from '../../generated/StableJoeStaking/Pair'
+
+function getJoePrice(): BigDecimal {
+  const pair = PairContract.bind(JOE_USDT_PAIR_ADDRESS)
+  const reservesResult = pair.try_getReserves()
+  if (reservesResult.reverted) {
+    log.info('[getJoePrice] getReserves reverted', [])
+    return BIG_DECIMAL_ZERO
+  }
+  const reserves = reservesResult.value
+  if (reserves.value0.toBigDecimal().equals(BigDecimal.fromString("0"))) {
+    log.error('[getJoePrice] USDT reserve 0', [])
+    return BIG_DECIMAL_ZERO
+  }
+  return reserves.value1.toBigDecimal().times(BIG_DECIMAL_1E18).div(reserves.value0.toBigDecimal()).div(BIG_DECIMAL_1E6)
+}
+
+export function convertAmountToDecimal(
+  tokenAmount: BigInt
+): BigDecimal {
+  return tokenAmount.toBigDecimal().div(exponentToBigDecimal(BigInt.fromString("18")));
+}
+
+export function exponentToBigDecimal(decimals: BigInt): BigDecimal {
+  let bd = BigDecimal.fromString("1");
+  for (
+    let i = BigInt.fromI32(0);
+    i.lt(decimals as BigInt);
+    i = i.plus(BigInt.fromI32(1))
+  ) {
+    bd = bd.times(BigDecimal.fromString("10"));
+  }
+  return bd;
+}
+
+export function handleClaimReward(event: ClaimRewardEvent): void {
+  const stableJoe = StableJoe.load(event.address.toHex())
+  let user = User.load(event.params.user.toHex())
+
+  if (!user) return
+
+  user.usdHarvested = user.usdHarvested.plus(convertAmountToDecimal(event.params.amount))
+  user.updatedAt = event.block.timestamp
+  
+  stableJoe.usdHarvested = stableJoe.usdHarvested.plus(convertAmountToDecimal(event.params.amount))
+  stableJoe.updatedAt = event.block.timestamp
+
+  stableJoe.save()
+
+  // update day data
+  const timestamp = event.block.timestamp.toI32()
+  const day = timestamp / 86400
+  const date = day * 86400
+  const id = event.address.toHex().concat('-').concat(BigInt.fromI32(day).toString())
+  let stableJoeDayData = StableJoeDayData.load(id);
+
+  if (USDT_ADDRESS === event.params.rewardToken || USDC_ADDRESS === event.params.rewardToken) {
```

Comment:
Only update usdHarvested if rewardToken is USDT or USDC, not sure if other stablecoins should be included

Reply:
USDC and USDC.e for now